octopusdeploy_framework/resource_team.go

```diff
@@ -0,0 +1,205 @@
+	r.Config = ResourceConfiguration(req, resp)
+}
+
+func (r *teamResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan schemas.TeamModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, "Creating team", map[string]interface{}{
+		"name": plan.Name.ValueString(),
+	})
+
+	team := mapTeamStateToResource(ctx, plan)
+	createdTeam, err := r.Client.Teams.Add(team)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating team", err.Error())
+		return
+	}
+
+	if err := r.updateUserRoles(ctx, plan, createdTeam); err != nil {
+		resp.Diagnostics.AddError("Error updating user roles for team", err.Error())
+		return
+	}
+
+	state := mapTeamResourceToState(ctx, createdTeam, plan, r.Client)
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+func (r *teamResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state schemas.TeamModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	team, err := r.Client.Teams.GetByID(state.ID.ValueString())
+	if err != nil {
+		if err := errors.ProcessApiErrorV2(ctx, resp, state, err, "teamResource"); err != nil {
+			resp.Diagnostics.AddError("unable to load team", err.Error())
+		}
+		return
+	}
+
+	newState := mapTeamResourceToState(ctx, team, state, r.Client)
+	resp.Diagnostics.Append(resp.State.Set(ctx, newState)...)
+}
+
+func (r *teamResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan schemas.TeamModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	team := mapTeamStateToResource(ctx, plan)
+	updatedTeam, err := r.Client.Teams.Update(team)
+	if err != nil {
+		resp.Diagnostics.AddError("Error updating team", err.Error())
+		return
+	}
+
+	if err := r.updateUserRoles(ctx, plan, updatedTeam); err != nil {
+		resp.Diagnostics.AddError("Error updating user roles for team", err.Error())
+		return
+	}
+
+	state := mapTeamResourceToState(ctx, updatedTeam, plan, r.Client)
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+func (r *teamResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state schemas.TeamModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := r.Client.Teams.DeleteByID(state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Error deleting team", err.Error())
+		return
+	}
+}
+
+func (*teamResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (r *teamResource) updateUserRoles(ctx context.Context, model schemas.TeamModel, team *teams.Team) error {
```

Comment:
As consequence of how the API updates user roles, user role state must be manually reconciled.